mvcc/watchable_store.go

```diff
@@ -180,6 +180,29 @@ func (s *watchableStore) cancelWatcher(wa *watcher) {
 	s.mu.Unlock()
 }
 
+func (s *watchableStore) Restore(b backend.Backend) error {
+	var err error
+	var needSync bool
+	func() {
+		s.mu.Lock()
+		defer s.mu.Unlock()
+		err = s.store.Restore(b)
+		if err != nil {
+			return
+		}
+
+		for wa := range s.synced.watchers {
+			s.unsynced.watchers.add(wa)
+		}
+		s.synced = newWatcherGroup()
+		needSync = true
+	}()
+	if needSync {
+		s.syncWatchers()
```

Comment:
`syncWatchersLoop` will pick this up; there'll already be a pause from restoring, so there's no much of an advantage calling it here